src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Resources/Patch/ConditionalPatchResourceHandlerTests.cs

```diff
@@ -0,0 +1,173 @@
+
+        _conditionalPatchHandler = new ConditionalPatchResourceHandler(
+            _searchService,
+            fhirDataStore,
+            conformanceProvider,
+            resourceWrapperFactory,
+            resourceIdProvider,
+            _authService,
+            _mediator,
+            logger);
+
+        // Setup search service to return one match
+        var searchResults = SearchResult.Empty();
+        searchResults.Results = new List<SearchResultEntry>
+        {
+            new SearchResultEntry(Samples.GetDefaultPatient(), SearchEntryMode.Match)
+        };
+        
+        _searchService.ConditionalSearchAsync(
+            Arg.Any<string>(),
+            Arg.Any<IReadOnlyList<Tuple<string, string>>>(),
+            Arg.Any<CancellationToken>(),
+            Arg.Any<ILogger>())
+            .Returns(searchResults);
+    }
+
+    [Fact]
+    public async Task GivenAConditionalPatchResourceHandler_WhenUserHasSearchAndUpdatePermissions_ThenPatchShouldSucceed()
+    {
+        // Arrange
+        _authService
+            .CheckAccess(DataActions.Read | DataActions.Write | DataActions.Search | DataActions.Update, CancellationToken.None)
+            .Returns(DataActions.Search | DataActions.Update);
+
+        var conditionalParameters = new List<Tuple<string, string>> { new("name", "John") };
+        var request = new ConditionalPatchResourceRequest("Patient", new FhirPathPatchPayload(new Parameters()), conditionalParameters, null);
+
+        // Act & Assert - Should not throw UnauthorizedFhirActionException
+        await _conditionalPatchHandler.Handle(request, CancellationToken.None);
+
+        await _mediator
+            .Received()
+            .Send<UpsertResourceResponse>(Arg.Any<UpsertResourceRequest>(), Arg.Any<CancellationToken>());
+    }
+
+    [Fact]
+    public async Task GivenAConditionalPatchResourceHandler_WhenUserHasLegacyReadAndWritePermissions_ThenPatchShouldSucceed()
+    {
+        // Arrange
+        _authService
+            .CheckAccess(DataActions.Read | DataActions.Write | DataActions.Search | DataActions.Update, CancellationToken.None)
+            .Returns(DataActions.Read | DataActions.Write);
+
+        var conditionalParameters = new List<Tuple<string, string>> { new("name", "John") };
+        var request = new ConditionalPatchResourceRequest("Patient", new FhirPathPatchPayload(new Parameters()), conditionalParameters, null);
+
+        // Act & Assert - Should not throw UnauthorizedFhirActionException
+        await _conditionalPatchHandler.Handle(request, CancellationToken.None);
+
+        await _mediator
+            .Received()
+            .Send<UpsertResourceResponse>(Arg.Any<UpsertResourceRequest>(), Arg.Any<CancellationToken>());
+    }
+
+    [Fact]
+    public async Task GivenAConditionalPatchResourceHandler_WhenUserHasOnlySearchPermission_ThenUnauthorizedExceptionIsThrown()
```

Comment:
Just a suggestion - should we club all the UnauthorizedFhirActionException facts together into a theory?